server/controllers/admin/suppliers.js

```diff
@@ -0,0 +1,132 @@
+/**
+* Supplier Controller
+*
+* This controller exposes an API to the client for reading and writing Supplier
+
+*/
+var db = require('../../lib/db');
+var uuid = require('../../lib/guid');
+// GET /Supplier
+//
+// The Supplier  is assumed from the session.
+exports.list = function list(req, res, next) {
+  'use strict';
+
+  var sql;
+  
+  sql =
+    'SELECT supplier.uuid, supplier.creditor_uuid, supplier.name, supplier.address_1, supplier.address_2, supplier.email, ' +
+    'supplier.fax, supplier.note, supplier.phone, supplier.international, supplier.locked ' +
+    'FROM supplier;';
+
+  if (req.query.locked === '0') {
+    sql =
+      'SELECT supplier.uuid, supplier.creditor_uuid, supplier.name, supplier.address_1, supplier.address_2, supplier.email, ' +
+      'supplier.fax, supplier.note, supplier.phone, supplier.international, supplier.locked ' +
+      'FROM supplier ' +
+      'WHERE supplier.locked = \'0\';';
```

Comment:
You do not need to escape hard-coded numbers in your SQL queries.  This
```js
supplier.locked = 0;
```
works fine.